enterprise/api/v2/views.py

```diff
@@ -0,0 +1,97 @@
+from rest_framework.status import HTTP_200_OK, HTTP_400_BAD_REQUEST
+from rest_framework_xml.renderers import XMLRenderer
+from six.moves.urllib.parse import quote_plus, unquote  # pylint: disable=import-error,ungrouped-imports
+
+from django.conf import settings
+from django.http import Http404
+from django.utils.decorators import method_decorator
+
+from enterprise import models
+from enterprise.api.v1.views import EnterpriseCustomerViewSet
+from enterprise.api.filters import EnterpriseCustomerUserFilterBackend, UserFilterBackend
+from enterprise.api.pagination import get_paginated_response
+from enterprise.api.throttles import ServiceUserThrottle
+from enterprise.api.v1 import serializers
+from enterprise.api.v1.decorators import enterprise_customer_required, require_at_least_one_query_parameter
+from enterprise.api.v1.permissions import HasEnterpriseEnrollmentAPIAccess, IsInEnterpriseGroup
+from enterprise.api_client.discovery import CourseCatalogApiClient
+from enterprise.constants import COURSE_KEY_URL_PATTERN
+
+LOGGER = getLogger(__name__)
+
+
+class EnterpriseCustomerViewSetV2(EnterpriseCustomerViewSet):
+    """
+    API views for the ``enterprise-customer`` API endpoint.
+    """
+
+    queryset = models.EnterpriseCustomer.active_customers.all()
+    serializer_class = serializers.EnterpriseCustomerSerializer
+
+    USER_ID_FILTER = 'enterprise_customer_users__user_id'
+    FIELDS = (
+        'uuid', 'slug', 'name', 'catalog', 'active', 'site', 'enable_data_sharing_consent',
+        'enforce_data_sharing_consent',
+    )
+    filter_fields = FIELDS
+    ordering_fields = FIELDS
+
+    @detail_route()
+    def courses(self, request, pk=None):  # pylint: disable=invalid-name,unused-argument
+        """
+        Retrieve the list of courses contained within the enterprise catalogs linked to this enterprise.
+
+        Only courses with active course runs are returned. A course run is considered active if it is currently
+        open for enrollment, or will open in the future.
+        """
+        enterprise_customer = self.get_object()
+        combined_content_filter = {}
+        for catalog in enterprise_customer.enterprise_customer_catalogs.all():
+            if not catalog.content_filter:
+                continue
+            update_content_filters(combined_content_filter, catalog.content_filter)
+
+        ######################### CONSTRUCTION ZONE #########################
+        # - Now that we have combined_content_filter, do something with it here
+        # - Call the discovery service using it
+        # - Get the results and serialize it
+        # (Looks like you can use something like get_catalog_courses, except while
+        #  accepting a query string. Need to figure out how to take this dict we created
+        #  and turn it into the proper datatype for the `querystring` variable)
+        catalog_api = CourseCatalogApiClient(request.user, enterprise_customer.site)
```

Comment:
so my original thought here was to use the `EnterpriseCustomerCatalogDetailSerializer` to both make the call and serialize the results by instantiating an EnterpriseCustomerCatalog with the new filter but not committing it to the database. I don't know 100% if that will work, but it would be preferable. in any case, the goal would be to reuse as much logic as possible for fetching and serializing responses from the discovery service search endpoint.